asadiqbal08/SOL-766: Share web certificate on Facebook

asadiqbal08 opened the pull request:
@mattdrayer @catong I just updated the flags as per discussion with @mattdrayer . Kindly look into the flag / text and let me know what you guys think. 

note: this is a WIP branch, commits will track my changes in this branch for SOL-766

mattdrayer commented:
@asadiqbal08 -- these changes look good -- my only question is around the DASHBOARD_TWITTER_TEXT -- @pbaruah, should this parameter control the text just for the dashboard sharing action, or should we just call it TWITTER_TEXT and use it for all Twitter sharing actions?

pbaruah commented:
@asadiqbal08, @mattdrayer - We should have separate settings for the example text for dashboard and for certificate sharing.

asadiqbal08 commented:
@mattdrayer  I will use the facebook button that @talbs introduced in pr #8649. waiting for this PR # #8649 to merge in master. Thanks

talbs commented:
@asadiqbal08, since my work is static (the links won't work with your magic), I'd rather not merge it into master, but rather have you cherry-pick the commit (ade0de4cad13d7255b330df8b5d920ee1e4a2406) onto this branch to help your work.

Make sense? Apologies for any confusion.

asadiqbal08 commented:
Jenkins, Test this please

asadiqbal08 commented:
Test this please

asadiqbal08 commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -22,6 +48,17 @@
+            picture: "${request.build_absolute_uri(course_image_url)}",
+            caption: 'Caption like which appear as title of the dialog box', // TODO# need to update as per requirement
+            description: 'Small description of the post' // TODO# need to update as per requirement
+        });
```
@pbaruah can you please suggest caption / description for this area ?

catong replied:
@asadiqbal08 @pbaruah is there a sandbox or a way to view how these strings would appear in context before we suggest wording? Thanks!

catong replied:
@asadiqbal08 Per conversation with @pbaruah can we test whether it's OK to leave caption and description empty?

asadiqbal08 replied:
@catong we can leave the caption empty. 
The description of the link (appears beneath the link caption). If not specified, this field is automatically populated by information scraped from the link, typically the title of the page.

catong replied:
Thanks, @asadiqbal08! I think this is more a decision for Product and Dev to make. @pbaruah @mattdrayer If I understand Asad correctly, if we do not define something for "Description" it is automatically populated -- I'm not too clear on what it gets populated with, or what the likely scraped info is going to be. Do you think that it's OK to also leave the description field value undefined and have the automatic population happen?

asadiqbal08 commented:
test this please

asadiqbal08 commented:
what's going wrong ? Jenkin test this please

asadiqbal08 commented:
Jenkin test this please

asadiqbal08 commented:
Jenkins test this please

mattdrayer commented:
@asadiqbal08 -- this PR is looking good -- once we have the caption/description from @pbaruah I think it's ready to go -- :+1: for me

mattdrayer commented:
@chrisndodge -- just looking for a second set of eyes on this PR -- if you can take a look that would be great.

chrisndodge commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -22,6 +48,17 @@
+            link: "${request.build_absolute_uri(reverse('cert_html_view', kwargs=dict(user_id=str(user.id),course_id=unicode(course_id))))}",
+            picture: "${request.build_absolute_uri(course_image_url)}",
+            caption: 'Caption like which appear as title of the dialog box', // TODO# need to update as per requirement
+            description: 'Small description of the post' // TODO# need to update as per requirement
```
I18N these strings (once you get final copy)

chrisndodge commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -10,6 +19,23 @@
         },
         dataType: 'json'
       });
+
```
Not crazy about having more JS inside the HTML template, can this go elsewhere?

mattdrayer replied:
@chrisndodge -- we're wondering if you have a suggestion of a reasonable location to move this Javascript?  There doesn't seem to be a dedicated spot for containing functions like this at the moment.   I suppose we could add a new facebook.js file to the mix -- thoughts on that?

catong commented:
@asadiqbal08 Apologies, I haven't had a chance yet to review the UI strings. I will take a look this morning.

catong commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -33,6 +70,26 @@ <h2 class="message-title hd-5 emphasized">${accomplishment_banner_opening}</h2>
                 <p class="message-copy copy copy-base emphasized">${accomplishment_banner_congrats}</p>
                 <div class="message-actions">
                     <h3 class="sr-only">${_("Take this with you:")}</h3>
```
@asadiqbal08 Can I view this in a sandbox, or can you provide a screenshot? I'd like to be clear on the context before I suggest wording. Thanks!

mattdrayer replied:
@catong -- if you catch up with @talbs he can probably provide you with a screenshot of the copy on the certificate page -- most of the strings you are commenting on were put in place by the UX team

catong replied:
@asadiqbal08 @mattdrayer  is there a sandbox?

mattdrayer replied:
![sol-766_facebook](https://cloud.githubusercontent.com/assets/4520595/8380011/6d25e142-1bf1-11e5-9778-9aa60bdeb96b.jpeg)

catong replied:
@asadiqbal08 @mattdrayer @frrrances My suggestion for this string: "Print or share your certificate."

catong commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -33,6 +70,26 @@ <h2 class="message-title hd-5 emphasized">${accomplishment_banner_opening}</h2>
+                      % if share_settings.get('CERTIFICATE_FACEBOOK', False):
+                        <a href="javascript:void(0)" onclick="shareOnFacebook('${facebook_share_text}'); return false;" class="action action-share-facebook btn btn-overlay btn-small icon-only" id="action-share-facebook">
+                            <i class="icon fa fa-facebook-official" aria-hidden="true"></i>
+                            <span class="action-label">${_("Post this on Facebook")}</span>
```
"Post this on Facebook" -> "Post on Facebook"

catong commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -33,6 +70,26 @@ <h2 class="message-title hd-5 emphasized">${accomplishment_banner_opening}</h2>
                     <h3 class="sr-only">${_("Take this with you:")}</h3>
+                    % if share_settings:
+                      <%
+                       facebook_share_text = _("I completed the {course_title} course on edX and here is the link to my certificate.").format(course_title=accomplishment_course_title)
```
@asadiqbal08 This text looks fine to me. 
Can a student replace this text?

asadiqbal08 replied:
student cannnot change this text but he/she can add its custom update on his/her feed

pbaruah commented:
@asadiqbal08 - Since it is not clear what the auto-populated scraped content will be, I would suggest splitting up the text and make the description a fixed line like- 
Main text: I completed the [course] on [platform name].
Description: Click on the link to see my certificate. 
Caption: Empty (should not show the text which shows up in grey color - CAPTION LIKE WHICH APPEAR AS TITLE OF THE D)

One question - what is the Yahoo image? Is that part of the student's FB profile?

cc: @mattdrayer @catong

asadiqbal08 commented:
@pbaruah @catong Thanks for the update. I updated the code as per suggestions, kindly look into that. 

@pbaruah I added the "yahoo image link" just for demonstration (In order to verify the look and feel of shared post into the user feed). Actually Facebook was not loading any image from local path (image from localhost environment). 

cc: @mattdrayer

catong commented on lms/templates/certificates/_accomplishment-banner.html:
```diff
@@ -22,6 +48,16 @@
+            name: text,
+            link: "${request.build_absolute_uri(reverse('cert_html_view', kwargs=dict(user_id=str(user.id),course_id=unicode(course_id))))}",
+            picture: "${request.build_absolute_uri(course_image_url)}",
+            description: "${_('Click on the link to see my certificate.')}"
```
@asadiqbal08 Sorry, one more small edit: "Click the link to see my certificate." (delete "on")

mattdrayer commented:
Jenkins, test this please.

mattdrayer commented:
Flaky test -- Jenkins, test this please.

mattdrayer commented:
Jenkins, test this please.

asadiqbal08 commented:
test this please

asadiqbal08 commented:
Hurry ! build is green @mattdrayer  :)
